addons/purchase_stock/models/purchase.py

```diff
@@ -511,7 +511,9 @@ def _prepare_stock_move_vals(self, picking, price_unit, product_uom_qty, product
             'date': date_planned,
             'date_deadline': date_planned,
             'location_id': self.order_id.partner_id.property_stock_supplier.id,
-            'location_dest_id': (self.orderpoint_id and not (self.move_ids | self.move_dest_ids)) and self.orderpoint_id.location_id.id or self.order_id._get_destination_location(),
+            'location_dest_id': (self.orderpoint_id and not (self.move_ids | self.move_dest_ids)
+                and (not picking or self.orderpoint_id.location_id in picking.location_dest_id.child_ids))
```

Comment:
Currently, this would break the following case (without any shenanigans with the picking types):
- Have a product with a RR on `WH/Stock/Shelf 2/Small Refrigerator` and a vendor set.
- Trigger the RR (through scheduler or manually).
- Confirm the generated PO.
- Both the picking and the move will point to `WH/Stock`, instead of having the move pointing towards `WH/Stock/Shelf 2/Small Refrigerator`.

Reply:
Thank you for the review @clesgow !

I have adapted the Fix and the test accordingly 😊